### DISCUSSION TITLE
Readonly fields

### DISCUSSION THREAD

**Discussion by rauni00** [pdfme/pdfme#509] [General]
Creating own schema and define which filed is readonly which one is not and when form Deginer open to edit then all fields are editable i want to resetrict that how to do ????

**Comment by vozmarkov** [comment]
Facing the same issue, any update from anyone ?

**Comment by peteward** [comment]
The readonly attribute of fields in pdfme is not intended to restrict what the Designer can do. It's used to prevent changing of fields within the Viewer and Form modes. 

To have it apply to the Designer it would probably need to consider roles so that users with fewer privileges could have restrictions in the Designer. 

This is a good feature request, but not supported currently.

**Comment by saajidjiffrey** [comment]
If anyone found an alternative solution, please lmk. 🙏🏻🫡
